### ISSUE TITLE
Extra Client Metadata in Registration Request throwing NPE

### ISSUE THREAD

**Issue by nynymike** [GluuFederation/oxAuth#1251]
When I add an extra param to the Registration Request, I am getting an NPE:

```
2020-02-10 04:03:12,653 ERROR [qtp392292416-10] [gluu.oxauth.register.ws.rs.RegisterRestWebServiceImpl] (RegisterRestWebServiceImpl.java:280) - null
java.lang.NullPointerException: null
```

This is contrary to the [Client Metadata Section of the Dynamic Client Registration](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata) spec which says:

```
Additional Client Metadata parameters MAY also be used. 
```

**Comment by yuriyz** [comment]
I tried request with custom parameters on ce-dev5.gluu.org where we have latest 4.1 and can't get NPE.
```
{
    "application_type":"web",
     "redirect_uris": [
         "https://client.example.org/callback",
          "https://client.example.org/callback2"
      ],
      "client_name": "Register Test Client 1",
      "subject_type": "pairwise",
      "token_endpoint_auth_method": "client_secret_basic",
      "custom1":"value1"
}
```
@nynymike would you please share your request?

**Comment by yuriyz** [comment]
Following NPE can be caught if send software statement without `jkws` or `jwks_uri` statment

```
2020-02-10 11:52:29,941 ERROR [qtp413601558-18] [gluu.oxauth.register.ws.rs.RegisterRestWebServiceImpl] (RegisterRestWebServiceImpl.java:290) - null
java.lang.NullPointerException: null
        at java.io.StringReader.<init>(StringReader.java:50) ~[?:1.8.0_222]
        at org.json.JSONTokener.<init>(JSONTokener.java:94) ~[json-20180813.jar:?]
        at org.json.JSONObject.<init>(JSONObject.java:399) ~[json-20180813.jar:?]
        at org.gluu.oxauth.register.ws.rs.RegisterRestWebServiceImpl.registerClientImpl(RegisterRestWebServiceImpl.java:133) ~[classes/:?]
``` 

Server return following error 
```
{"reason":"Unknown.","error_description":"The value of one of the Client Metadata fields is invalid and the server has rejected this request. Note that an Authorization Server MAY choose to substitute a valid value for any requested parameter of a Client's Metadata.","error":"invalid_client_metadata"}
```

We can return back more specific error.

**Comment by yuriyz** [comment]
Add description of `software_statement` to docs (including https://gluu.org/docs/ce/4.0/api-guide/openid-connect-api/#openid-connect-register-client-api)

**Comment by yuriyz** [comment]
Missed API parameters in docs should be covered by @shmorri in this ticket  https://github.com/GluuFederation/docs-ce-prod/issues/227

**Comment by yuriyz** [comment]
Fixed in 4.1 and 4.2.
